File: #4/alter.js
const alter = (n) => {
  var aux = -1
  const xs = Array.from({length: n*2}, (_, i) => i + 1)
  const ys = Array.from({length: n*2}, (_, i) => i = aux--)
  
  return intercal(xs, ys)
}

const intercal = (xs, ys) => {
  if(xs.length == 0){
    return ys
  }else if(ys.length == 0) {
    return xs
  }else {
    return [xs[0], ys[0]].concat(intercal(xs.slice(1), ys.slice(1)))   
  }
}

console.log(alter(2))

File: #4/alter.hs
alter a = intercal xs ys
  where 
    xs = [1..a]
    ys = [x * (-1) | x <- [1..a]]

intercal xs [] = xs
intercal [] ys = ys
intercal (x:xs) (y:ys) = [x,y] ++ intercal xs ys

main = do
  a <- readLn :: IO Int
  print $ alter a